crates/ruma-state-res/src/event_auth.rs

```diff
@@ -500,12 +502,15 @@ fn valid_membership_change(
         // Is the authorised user allowed to invite users into this room
         let (auth_user_pl, invite_level) = if let Some(pl) = &power_levels_event {
             // TODO Refactor all powerlevel parsing
-            let invite = match from_json_str::<PowerLevelsContentInvite>(pl.content().get()) {
-                Ok(power_levels) => power_levels.invite,
-                _ => int!(50),
-            };
+            let invite =
+                match deserialize_power_levels_content_invite(pl.content().get(), room_version) {
+                    Ok(power_levels) => power_levels.invite,
+                    _ => int!(0),
+                };
 
-            if let Ok(content) = from_json_str::<PowerLevelsContentFields>(pl.content().get()) {
+            if let Ok(content) =
+                deserialize_power_levels_content_fields(pl.content().get(), room_version)
```

Comment:
Same here, the `else` branch falls back to some default value if deserialization fails.